ksql-streams/src/main/java/io/confluent/ksql/execution/streams/RoutingFilter.java

```diff
@@ -25,10 +26,34 @@
  */
 public interface RoutingFilter {
 
-  boolean filter(
-      HostInfo activeHost,
-      KsqlHost hostInfo,
-      String storeName,
-      int partition);
+  /**
+   * If the given host should be included for consideration.
+   * @param activeHost The active host
+   * @param hostInfo The host to be considered for this filter
+   * @return If the host should be included for consideration.
+   */
+  boolean filter(HostInfo activeHost, KsqlHost hostInfo);
+
+  /**
+   * A factory for RoutingFilters.
+   */
+  interface RoutingFilterFactory {
+
+    /**
+     * Creates a RoutingFilter
+     * @param routingOptions The options to use when filtering
+     * @param hosts The set of all hosts that have the store, including actives and standbys
+     * @param applicationQueryId The application query id
+     * @param storeName The state store name
+     * @param partition The partition of the changelog topic
+     * @return The new RoutingFilter
+     */
+    RoutingFilter createRoutingFilter(
+        RoutingOptions routingOptions,
+        List<KsqlHost> hosts,
```

Comment:
could we just move the `activeHost` also here? leaving just the 1 argument for `filter()`

Reply:
Yeah, we can.  I agree with that.  It's also not set per filter call and makes the API a bit confusing being set with the other host.